### PR TITLE
Enable github actions for llvm

### PR DESCRIPTION
**Pull request by michalpasztamobica**
The action will build llvm according to [flang build instructions](https://github.com/flang-compiler/flang/wiki/Building-Flang) and upload the resulting artifacts for subsequent flows to use.
The purpose is to be able to build [flang-driver](https://github.com/flang-compiler/flang-driver/pull/94) and [flang](https://github.com/flang-compiler/flang/pull/943) in subsequent steps.
Optionally, we can also enable this action to run on PRs, but it takes about 1 hour to complete the build.
An example of a complete action can be viewed [here](https://github.com/michalpasztamobica/llvm/actions/runs/320389799).

FYI, @RichBarton-Arm

### CODE REVIEW

**Review by RichBarton-Arm — APPROVED** [review]
This all LGTM from a quick look. We have examples of this running well in Michal's fork so I am happy to approve on that basis.

**Comment by gklimowicz** [comment]
I don't have an environment set up right now to evaluate this. I am removing myself from the list of reviewers to keep from holding this up.

File: .github/workflows/build_llvm.yml
Comment (bryanpkc) [inline .github/workflows/build_llvm.yml:30]: Question: Why does the GitHub action put the process inside `/home/runner/work/llvm/llvm` on start-up?

File: .github/workflows/build_llvm.yml
Comment (bryanpkc) [inline .github/workflows/build_llvm.yml:46]: What is the  installation path? It's not clear since `CMAKE_INSTALL_PREFIX` is not set.

File: .github/workflows/build_llvm.yml
Comment (michalpasztamobica) [reply .github/workflows/build_llvm.yml:30]: Checkout action [README](https://github.com/actions/checkout) says:

> Improved layout
> * The input path is always relative to $GITHUB_WORKSPACE
> * Aligns better with container actions, where $GITHUB_WORKSPACE gets mapped in

I tried to set `path` to `../..` but got an error, so the `cd ../..` is a workaround for checkout's limitation.

Github actions are developing and growing quite quickly, so perhaps soon there will be a way to simplify this.

File: .github/workflows/build_llvm.yml
Comment (michalpasztamobica) [reply .github/workflows/build_llvm.yml:46]: This was also a point raised by @banach-space.

The default installation path on Ubuntu is `/usr/local`, but actually I didn't care much as long as it is the same for all cmake scripts because our workspace is temporary and gets destroyed after every job run. 

You're right - we could explicitly set it with the macro, but this would require repeating the set up for every command and would make the scripts less readable and longer. Every `step` runs in a new shell and has to set up `CMAKE_OPTIONS` anew.

I think defaults are there for us to use them and they are set up in a reasonable and consequent way. So if we don't care about the path in this case, I though it's best to keep the scripts simple...

Comment (banach-space) [reply .github/workflows/build_llvm.yml:46]: I'm always in favor of being explicit with these things.

Today the installation path seems obvious as we've discussed it recently. However, we also want to make sure that:
* it is obvious to us in not-so-distant future
* it is obvious to anyone who hasn't participated in this PR

In the current set-up we rely on people knowing what the default is. By setting `CMAKE_INSTALL_PREFIX` we no longer rely on that knowledge. Instead, we define and document what that path is. So yes, my preference would be to set `CMAKE_INSTALL_PREFIX`. Even if initially it's set to `/usr/local`.

@michalpasztamobica Thank you for working on this!

Comment (michalpasztamobica) [reply .github/workflows/build_llvm.yml:46]: @bryanpkc, @banach-space, thanks for you remarks!

I just pushed an update to all three projects including your suggestion to be explicit about the path and also some more slight improvements:
* I added a scheduled build once every second month - this is to make sure we have artifacts in place even if there would be no pushes to the repo for more than 90 days (unlikely, but better be sure)
* I packaged the build directory along with source code, to avoid the situation where versions don't match (thanks for the suggestion, @bryankpc)

The flow is still working, see [llvm build](https://github.com/michalpasztamobica/llvm/actions/runs/337486934), [flang-driver build](https://github.com/michalpasztamobica/flang-driver/actions/runs/337592116) and [flang build](https://github.com/michalpasztamobica/flang/actions/runs/337689520) .

I placed changes in a separate commit to make the review easier. I'll squash and rename the project to `flang-compiler` once the review is approved.

File: .github/workflows/build_llvm.yml
Comment (bryanpkc) [reply .github/workflows/build_llvm.yml:30]: Thanks @michalpasztamobica.

**Review by bryanpkc — APPROVED** [review]
LGTM.

**Comment by michalpasztamobica** [comment]
I pushed squashed branch. @shivaramaarao , any comments from your side on this or the other two related PRs?

**Review by shivaramaarao — APPROVED** [review]

**Comment by kiranchandramohan** [comment]
@michalpasztamobica I see two commits. Are these two necessary or do you have to squash again?

**Comment by michalpasztamobica** [comment]
You're right, @kiranchandramohan , I did something wrong here, now there's just one squashed commit.

**Comment by michalpasztamobica** [comment]
Without this PR merged, the other two (flang-driver and flang) are failing due to missing artifacts. @kiranchandramohan , once this gets in and new artifacts show up in `Actions` tab, would you be able to rerun the failed jobs in [flang-drive](https://github.com/flang-compiler/flang-driver/actions/runs/344164353)r and [flang](https://github.com/flang-compiler/flang/actions/runs/344165581), please? There should be a "Re-run jobs" button available to repository maintainers in the upper right corner.
Thanks in advance!

**Pull request merged**
